components/clipboard-button/index.js

```diff
@@ -83,8 +83,17 @@ class ClipboardButton extends Component {
 		const classes = classnames( 'components-clipboard-button', className );
 		const ComponentToUse = icon ? IconButton : Button;
 
+		// Workaround for inconsistent behavior in Safari, where <textarea> is not
+		// the document.activeElement at the moment when the copy event fires.
+		// This causes documentHasSelection() in the copy-handler component to
+		// mistakenly override the ClipboardButton, and copy a serialized string
+		// of the current block instead.
```

Comment:
Thanks for the detailed comment, explaining why do we have the workaround.

Reply:
One problem in Safari, though, is that the while block (`<!-- p:file {"href":…`) ends up in the clipboard.

Reply:
Ok, it looks like this workaround hadn't fixed the Safari bug completely. I did some testing and narrowed down the repro condition a bit. My workaround seems to be effective for pre-existing posts (i.e. posts that were already saved when the page loaded), but not for new posts. Not sure what's going on, but that's a start... I'll have to look into it further.

Reply:
Addressed in #7106